### PR TITLE
fix for returning incorrect builder in hybrid relations

### PR DESCRIPTION
**Pull request by pimlie**
@jenssegers @Tucker-Eric 

Commit 266305caeed70cf8ee787bcd0140c9fca2f0bcd6 introduced a bug when using HybridRelations in combination with e.g. raw methods. The `newEloquentBuilder` method always returned a `Illuminate\Database\Eloquent\Builder` but for `Jenssegers\Mongodb\Eloquent\Model` models it should return a `Jenssegers\Mongodb\Eloquent\Builder`.

### CODE REVIEW

**Comment by pimlie** [comment]
@jenssegers Sorry for the failed test, I have fixed the CI and now it passes. I would be obliged if you could merge this :)

File: tests/models/User.php [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 use Jenssegers\Mongodb\Eloquent\Model as Eloquent;
+use Jenssegers\Mongodb\Eloquent\HybridRelations;
 use Illuminate\Auth\Authenticatable;
 use Illuminate\Auth\Passwords\CanResetPassword;
 use Illuminate\Contracts\Auth\Authenticatable as AuthenticatableContract;
 use Illuminate\Contracts\Auth\CanResetPassword as CanResetPasswordContract;
 
 class User extends Eloquent implements AuthenticatableContract, CanResetPasswordContract
 {
-    use Authenticatable, CanResetPassword;
```
Comment (jenssegers) [inline tests/models/User.php]: This should not be needed right? Since `Eloquent` had this trait as well?

Comment (pimlie) [reply tests/models/User.php]: Not sure what you mean, but I added the `HybridRelations` trait for the User model as using that trait immediately gives an error without this fix. In the first commit I used 2 `use` statements which I replaced to be 1 statement, therefor this line shows as deleted.

I dont think the Eloquent model class (either mongodb's or laravel's) implements one of these traits already

Comment (jenssegers) [reply tests/models/User.php]: `Jenssegers\Mongodb\Eloquent\Model` already has the `HybridRelations` trait, which the model extends from.

Comment (pimlie) [reply tests/models/User.php]: Ah ok. Thanks for merging.

**Pull request merged**